Entities/Attachable.cpp

```diff
@@ -1,679 +1,447 @@
-            extruded = m_Pos - extruded;
-            g_SceneMan.WrapPosition(extruded);
-            pParentActor->AlarmPoint(extruded);
-        }
-    }
-
-    return penetrated;
-}
+		m_InheritsHFlipped = reference.m_InheritsHFlipped;
+		m_InheritsRotAngle = reference.m_InheritsRotAngle;
+		m_InheritedRotAngleOffset = reference.m_InheritedRotAngleOffset;
 
+		m_CollidesWithTerrainWhileAttached = reference.m_CollidesWithTerrainWhileAttached;
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  GibThis
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Gibs this, effectively destroying it and creating multiple gibs or
-//                  pieces in its place.
-
-void Attachable::GibThis(Vector impactImpulse, float internalBlast, MovableObject *pIgnoreMO) {
-    if(m_pParent){
-      m_pParent->RemoveAttachable(this);
-    }
-    else{
-      Detach();
-    }
-    MOSRotating::GibThis(impactImpulse, internalBlast, pIgnoreMO);
-}
+		m_AtomSubgroupID = GetUniqueID();
 
+		return 0;
+	}
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  Attach
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Attaches this Attachable to a host MOSprite
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int Attachable::ReadProperty(std::string propName, Reader &reader) {
+		if (propName == "ParentOffset") {
+			reader >> m_ParentOffset;
+		} else if (propName == "DrawAfterParent") {
+			reader >> m_DrawAfterParent;
+		} else if (propName == "DeleteWhenRemovedFromParent") {
+			reader >> m_DeleteWhenRemovedFromParent;
+		} else if (propName == "ApplyTransferredForcesAtOffset") {
+			reader >> m_ApplyTransferredForcesAtOffset;
+		} else if (propName == "GibWithParentChance") {
+			reader >> m_GibWithParentChance;
+		} else if (propName == "ParentGibBlastStrengthMultiplier") {
+			reader >> m_ParentGibBlastStrengthMultiplier;
+		} else if (propName == "JointStrength" || propName == "Strength") {
+			reader >> m_JointStrength;
+		} else if (propName == "JointStiffness" || propName == "Stiffness") {
+			reader >> m_JointStiffness;
+		} else if (propName == "JointOffset") {
+			reader >> m_JointOffset;
+		} else if (propName == "BreakWound") {
+			m_BreakWound = dynamic_cast<const AEmitter *>(g_PresetMan.GetEntityPreset(reader));
+			if (!m_ParentBreakWound) { m_ParentBreakWound = m_BreakWound; }
+		} else if (propName == "ParentBreakWound") {
+			m_ParentBreakWound = dynamic_cast<const AEmitter *>(g_PresetMan.GetEntityPreset(reader));
+		} else if (propName == "InheritsHFlipped") {
+			reader >> m_InheritsHFlipped;
+			if (m_InheritsHFlipped != 0 && m_InheritsHFlipped != 1) { m_InheritsHFlipped = 2; }
```

Comment:
Reverse HFlip should be -1.

Reply:
Hmmm I'm on the fence about this a bit. It lines up with FlipNumber and stuff, but it doesn't work if this turns into an enum. Let's chat in source and maybe see what other folks think.

Reply:
Gonna use -1 for this as discussed